src/libasr/pass/unique_symbols.cpp

```diff
@@ -240,6 +277,12 @@ class SymbolRenameVisitor: public ASR::BaseWalkVisitor<SymbolRenameVisitor> {
             ASR::symbol_t *sym = ASR::down_cast<ASR::symbol_t>((ASR::asr_t*)&x);
             sym_to_renamed[sym] = update_name(x.m_name);
         }
+        if ( c_mangling ) {
+            if(is_restricted_c(x.m_name)){
+                ASR::symbol_t *sym = ASR::down_cast<ASR::symbol_t>((ASR::asr_t*)&x);
+                sym_to_renamed[sym] = "_xx_"+std::string(x.m_name)+"_xx_";
```

Comment:
Let's put these into a function, so that there is just one place that decides how to mangle the C names.